### PR TITLE
Add OracleAzureDbPostgreSQLSync scenario and ServiceTask resource

### PR DESCRIPTION
**Pull request by huang91shu**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by adxsdknet** [comment]
# Automation for azure-sdk-for-net 
A PR has been created for you: 
https://github.com/azure/azure-sdk-for-net/pull/6002 
.NET SDK Commits: 
https://github.com/adxsdknet/azure-sdk-for-net/pull/6002/commits/47e8f9592f9a3dfe905afd19c2c3bafbc9df3ba0

File: specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/datamigration.json [hunk 1]
```diff
@@ -751,6 +800,163 @@
...
+            "schema": {
+              "$ref": "./definitions/Tasks.json#/definitions/ProjectTask"
+            }
+          },
+          "default": {
+            "description": "Error",
+            "schema": {
+              "$ref": "./definitions/Common.json#/definitions/ApiError"
+            }
+          }
```
Comment (dsgouda) [inline specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/datamigration.json]: Is this a long running operation?

File: specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/datamigration.json [hunk 2]
```diff
@@ -751,6 +800,163 @@
...
+          "Tasks_Delete": { "$ref": "./examples/ServiceTasks_Delete.json" }
+        },
+        "responses": {
+          "200": {
+            "description": "Service Task resource deleted"
+          },
+          "204": {
+            "description": "Not found"
+          },
+          "default": {
```
Comment (dsgouda) [inline specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/datamigration.json]: Please check if this is a long running operation

File: specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/datamigration.json [hunk 3]
```diff
@@ -800,6 +1006,52 @@
...
+            "$ref": "#/parameters/serviceName"
+          },
+          {
+            "$ref": "#/parameters/taskName"
+          },
+          {
+            "$ref": "#/parameters/apiVersion"
+          }
+        ],
+        "responses": {
```
Comment (dsgouda) [inline specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/datamigration.json]: Please check if this is a long running operation

File: specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/definitions/ServiceFeatureOCITask.json [hunk 4]
```diff
@@ -0,0 +1,142 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Azure Database Migration Service Resource Provider",
+    "version": "2018-07-15-preview"
```
Comment (dsgouda) [inline specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/definitions/ServiceFeatureOCITask.json]: Are all these definitions part of this version or have they been copied over from an older version?

**Review by dsgouda — CHANGES_REQUESTED** [review]

File: specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/datamigration.json (hunk 1 above)
Comment (huang91shu) [reply specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/datamigration.json]: Yes, same as project task.  They behave the same just that we created a new resource under a different endpoint.

File: specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/datamigration.json (hunk 2 above)
Comment (huang91shu) [reply specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/datamigration.json]: Yes, same as project task.  We queue up the work which gets picked up by a worker.

File: specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/datamigration.json (hunk 3 above)
Comment (huang91shu) [reply specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/datamigration.json]: See above.

File: specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/definitions/ServiceFeatureOCITask.json (hunk 4 above)
Comment (huang91shu) [reply specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/definitions/ServiceFeatureOCITask.json]: These are new features which were not available before.  We have no breaking changes so I kept the version.

File: specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/datamigration.json (hunk 1 above)
Comment (hitenjava) [reply specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/datamigration.json]: These tasks are not tracked resources.
Similar to project task, they are different from the long running operation defined for ARM tracked resource.
(ARM tracked resource long running operations return Azure-AsyncOperation which the client has to query for status - For project and service tasks, we do not follow this.)

**Review by dsgouda — CHANGES_REQUESTED** [review]
Looks good for the most part left a couple of comments.
SInce these are not ARM resource operations, we can skip the review

File: specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/definitions/Tasks.json [hunk 5]
```diff
@@ -609,7 +681,7 @@
         }
       ]
     },
-	"ValidateMigrationInputSqlServerSqlMISyncTaskProperties": {
+	  "ValidateMigrationInputSqlServerSqlMISyncTaskProperties": {
```
[the remark this answers is not in the document]
Comment (huang91shu) [reply specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/definitions/Tasks.json]: This was to fix the indentation.  Previously it was one tab too few.  It doesn't seem to show correctly when comparing.

File: specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/definitions/ServiceFeatureOCITask.json [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+      }
+    }
+  },
+  "paths": {},
+  "definitions": {
+    "OracleOCIDriverInfo": {
+      "type": "object",
+      "description": "Information about an Oracle OCI driver.",
+      "properties": {
+        "driverName": {
```
[the remark this answers is not in the document]
Comment (huang91shu) [reply specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/definitions/ServiceFeatureOCITask.json]: Yup, this object is only returned so all properties should be readonly.

File: specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/definitions/ServiceFeatureOCITask.json [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+        "installedDriver": {
+          "type": "object",
+          "description": "Information about the installed driver if found and valid.",
+          "additionalProperties": {
+            "type": "array",
+            "items": {
+              "$ref": "#/definitions/OracleOCIDriverInfo"
+            }
+          },
+          "readOnly": true
```
[the remark this answers is not in the document]
Comment (huang91shu) [reply specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/definitions/ServiceFeatureOCITask.json]: I think it's the indentation or order that is confusing.  The readOnly property is on the installedDriver property.  We many multiple examples of this same structure.

File: specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/definitions/ServiceFeatureOCITask.json (hunk 6 above)
[the remark this answers is not in the document]
Comment (dsgouda) [reply specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/definitions/ServiceFeatureOCITask.json]: In that case readonly should be set on each property individually

**Comment by huang91shu** [comment]
Does anyone know how I can find what the error in the automated public.rest-api-specs failure is?  I looked at the log but don't see anything wrong in the json.  I ran the file through a validator and it seems to be correct too.

The error is "Path 'paths['/subscriptions/{subscriptionId}/resourceGroups/{groupName}/providers/Microsoft.DataMigration/services/{serviceName}'].parameters', line 130, position 21."  Cannot deserialize the current JSON array.

It seems to be complaining about the parameters for that path but the line number doesn't make sense.  The parameters field is on line 107 and seems to formatted correctly.

**Comment by dsgouda** [comment]
@sarangan12 @sergey-shandar could you please take a look at the failures [here](https://dev.azure.com/azure-sdk/public/_build/results?buildId=32517&view=logs)

**Comment by sergey-shandar** [comment]
@dsgouda could you reaise a bug for the diff tool https://github.com/Azure/openapi-diff/issues ?

**Comment by dsgouda** [comment]
@sergey-shandar I have approved the PR, if the build failures are bugs, please merge this

**Review by dsgouda — APPROVED** [review]
LGTM

**Comment by huang91shu** [comment]
I've added back all my original changes.  I'm ready for merging other than the one failing test which we can't figure out.  Thank you for the help looking into this.

**Comment by vchske** [comment]
Hello, I'm wondering if we could get eyes on this. There are some Powershell and CLI deliverables that are backed up behind this. Thanks!

**Pull request merged**